src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxReceiverInputNode.cs

```diff
@@ -65,20 +76,30 @@ public Builder(SyntaxReceiverInputNode owner, DriverStateTable driverStateTable)
 
             public ISyntaxInputNode SyntaxInputNode { get => _owner; }
 
+            [MemberNotNullWhen(true, nameof(_inputSteps))]
+            private bool TrackIncrementalSteps => _nodeStateTable.TrackIncrementalSteps;
```

Comment:
>_nodeStateTable.TrackIncrementalSteps [](http://example.com/codeflow?start=50&length=37)

If `TrackIncrementalSteps` is tied to `_nodeStateTable.TrackIncrementalSteps`, did we need the `trackIncrementalSteps` parameter in the constructor?

Reply:
We produce the value for `_nodeStateTable` in the constructor based partially on the `trackIncrementalSteps` parameter in the constructor.